overlord/auth/auth.go

```diff
@@ -183,3 +183,41 @@ NextUser:
 	}
 	return nil, ErrInvalidAuth
 }
+
+// An AuthContext handles user updates.
+type AuthContext interface {
+	UpdateUser(user *UserState) error
+}
+
+// AuthStateContext helps keeping track and updating users in the state.
+type AuthStateContext struct {
```

Comment:
Let's please define this as `type authContext struct` and make it private to this package. Externally it'll be known via its interface only.